src/utils.cc

```diff
@@ -184,3 +188,35 @@ std::vector<std::string> utils::SpiltWithSymbol(const std::string& str,
   free(input);
   return str_list;
 }
+std::string utils::GetCommandExecuteResult(const std::string& command) {
+  std::string result;
+  CommandExecutor commandExecutor;
+  int save_fd_out = dup(STDOUT_FILENO);
+  int save_fd_err = dup(STDERR_FILENO);
+  std::string temporary_file = "file.txt";
+  int fd_out =
+      open(temporary_file.c_str(), O_WRONLY | O_APPEND | O_CREAT, 0664);
+  dup2(fd_out, STDOUT_FILENO);
+  dup2(fd_out, STDERR_FILENO);
+  close(fd_out);
+  commandExecutor.Execute(command);
+  std::ifstream fin(temporary_file);
+  while (fin >> result) {
+  }
+  dup2(save_fd_out, STDOUT_FILENO);
+  dup2(save_fd_err, STDERR_FILENO);
+  remove(temporary_file.c_str());
+  return result;
+}
+
+std::string utils::GetBranchName(std::string prompt_line) {
+  auto branch_name = utils::GetCommandExecuteResult(
+      R"(git branch | grep "^\*" | sed 's/^..//')");
+  auto is_git_manage = utils::GetCommandExecuteResult(
+                           "git rev-parse --is-inside-work-tree") == "true";
+
+  if (!branch_name.empty() && branch_name != ".git" && is_git_manage) {
+    prompt_line += "\033[31m git(" + branch_name + ")\033[0m";
```

Comment:
What does `\033[31m` means? Can we use some macro or const variable to make it more readable?